include/dlaf/matrix/matrix.h

```diff
@@ -23,9 +23,27 @@
 #include "dlaf/matrix/tile.h"
 #include "dlaf/types.h"
 
+#include "dlaf/common/range2d.h"
+
 namespace dlaf {
 namespace matrix {
 
+namespace internal {
+
+/// Helper function returning a vector with the results of calling a function over a IterableRange2D
+///
+/// @param f non-void function accepting LocalTileIndex as parameter
+template <class Func>
+auto selectGeneric(Func&& f, common::IterableRange2D<SizeType, LocalTile_TAG> range) {
+  using RetT = decltype(f(LocalTileIndex{}));
+
+  std::vector<RetT> tiles;
```

Comment:
To avoid potentially avoid some `malloc` calls, I suggest to preallocate the vector :`tiles.reserve()`.

Reply:
yes, I force pushed the updated branch where I had already fixed this (sorry for that, it is because this change belongs to another PR which this one depends onto)